### PR TITLE
codeintel: Remove tables from api docs

### PR DESCRIPTION
**Pull request by efritz**
Remove all db-level objects related to api docs.

## Test plan

CI is running main-dry-run branches.

### CODE REVIEW

File: migrations/codeintel/1665531314_remove_api_docs_tables/up.sql [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+DROP TABLE IF EXISTS lsif_data_documentation_mappings CASCADE;
+
+DROP TABLE IF EXISTS lsif_data_documentation_pages CASCADE;
+
+DROP TABLE IF EXISTS lsif_data_documentation_path_info CASCADE;
+
+-- DROP FUNCTION IF EXISTS get_file_extension(text);
+-- DROP FUNCTION IF EXISTS path_prefixes(text);
+-- DROP FUNCTION IF EXISTS singleton(text);
+-- DROP FUNCTION IF EXISTS singleton_integer(integer);
```
Comment (eseliger) [inline migrations/codeintel/1665531314_remove_api_docs_tables/up.sql:68]: what about this?

**Review by eseliger — APPROVED** [review]

File: migrations/codeintel/1665531314_remove_api_docs_tables/up.sql (hunk 1 above)
Comment (efritz) [reply migrations/codeintel/1665531314_remove_api_docs_tables/up.sql:68]: They had dependencies, but I rearranged some stuff. I'll see if there's still an ordering issue and if so we can do it in the next release.

Comment (eseliger) [reply migrations/codeintel/1665531314_remove_api_docs_tables/up.sql:68]: cool, as long as they're not forgotten I'm good with that.

File: migrations/frontend/1665588249_rename_bad_fk_constraint/up.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ALTER TABLE
+    lsif_uploads_reference_counts DROP CONSTRAINT IF EXISTS lsif_uploads_reference_counts_upload_id_fk;
+
+ALTER TABLE
+    lsif_uploads_reference_counts DROP CONSTRAINT IF EXISTS lsif_data_docs_search_private_repo_name_id_fk;
+
+ALTER TABLE
+    lsif_uploads_reference_counts
+ADD
+    CONSTRAINT lsif_uploads_reference_counts_upload_id_fk FOREIGN KEY (upload_id) REFERENCES lsif_uploads(id) ON DELETE CASCADE;
```
Comment (chrismwendt) [inline migrations/frontend/1665588249_rename_bad_fk_constraint/up.sql:11]: It doesn't look like the up/down scripts are inverses of each other. Both drop 2 constraints and add 1, which doesn't bring the DB back to the same state.

**Review by chrismwendt — APPROVED** [review]
You might want to double check the up/downs I commented on in case there's a problem with them.

File: migrations/frontend/1665588249_rename_bad_fk_constraint/up.sql (hunk 2 above)
Comment (efritz) [reply migrations/frontend/1665588249_rename_bad_fk_constraint/up.sql:11]: Initial state: we have *badname*
Run up, then we drop *badname* and *goodname* (no-op), then make *goodname*
Run down, then we drop *badname* (no-op) and *goodname* and then make *badname*

File: migrations/codeintel/1665531314_remove_api_docs_tables/up.sql (hunk 1 above)
Comment (efritz) [reply migrations/codeintel/1665531314_remove_api_docs_tables/up.sql:68]: lol they're all rockskip related things. fixed

**Pull request merged**